blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx

```diff
@@ -80,10 +82,12 @@ class RunDetailsHeader extends Component {
             </h1>
         );
 
+        const branchUrl = `${buildPipelineUrl(run.organization, pipeline.fullName)}/activity/${run.pipeline}`;
+
         const branchSourceDetails = (
             <div className="u-label-value" title={branchLabel + ': ' + displayName}>
                 <label>{ branchLabel }:</label>
-                <span>{ displayName }</span>
+                <Link to={ branchUrl }>{ displayName }</Link>
```

Comment:
not sure how, but if this is a link, ideally it would only show when it is multibranch